src/utils/index.ts

```diff
@@ -1,7 +1,66 @@
+import { ethers } from "ethers";
+
 export function getPageCoords(element: HTMLElement | SVGElement) {
-  const { top, left } = element.getBoundingClientRect();
+  const {
+    top,
+    left,
+    right,
+    bottom,
+    width,
+    height,
+  } = element.getBoundingClientRect();
   return {
     top: window.innerHeight - top,
-    left: left,
+    left,
+    right,
+    bottom,
+    width,
+    height,
   };
 }
+
+export function updateDefaultObject(
+  defaultObj: Record<
+    string,
+    {
+      value: any | undefined;
+      [k: string]: unknown;
+    }
+  >,
+  freshObj: Record<
+    string,
+    {
+      value: any;
+      [k: string]: unknown;
```

Comment:
since you dont care about any other values than `value` on fresh obj, you could slightly simplify the structure to just `{[string]: any}` and assume that `value` going into default obj. obviously though if you plan to update keys other than value the way you have it is best